src/main/java/net/reini/rabbitmq/cdi/EventBinder.java

```diff
@@ -297,6 +300,18 @@ public ExchangeBinding withRoutingKey(String routingKey) {
       LOGGER.info("Routing key for event type {} set to {}", eventType.getSimpleName(), routingKey);
       return this;
     }
+    
+    /**
+     * Sets the message converter to be user for message publishing.
+     * 
+     * @param messageConverter The message converter 
+     * @return the exchange binding
+     */
+    public ExchangeBinding withConverter(MessageConverter messageConverter){
```

Comment:
Here we should convert the `ExchangeBinding` to a generic variant of `bind(Class<M> messageType)` to allow verification of the generic `MessageConverter<M>`.  This I would rename to `MessageEncoder<M>`  to have `MessageDecoder<M>` the receiving end.